File: 0217-contains-duplicate/0217-contains-duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 [1,2,3,1]
    {1: 2, 2: 1, 3: 1}
    if value > 1
        return True
    else 
        False
 - loop through to check all values
 - 
 */
var containsDuplicate = function(nums) {
    let unique_map = new Map()

    for (let num of nums){ // [1,2,3,1] {1: 2, 2: 1, 3:1}
        unique_map.set(num, (unique_map.get(num) || 0) + 1)
    }

    for(let i = 0; i < nums.length; i++){ // {1: } [2,14,18,22,22]
        if(unique_map.get(nums[i]) > 1){
            return true
        } 
    }

    return false
};

File: 0217-contains-duplicate/0217-contains-duplicate.py
"""
[1,2,3,1] -> true
{1: 1, 2: 1, 3: 1, 1 return false}

[1,2,3,4]
"""
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        nums_map = {} # {1: 1, 2: 1, 3: 1, 4: 1}

        for i, element in enumerate(nums): # {}
            if element not in nums_map:
                nums_map[element] = 1
            else:
                return True
        
        return False
        